app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -52,21 +62,100 @@ public void onClick(final View view) {
 
         final ListView lv = (ListView) findViewById(R.id.listView);
 
-        //Retrieve the contacts from the database
-        getContacts();
+        final Intent intent = getIntent();
+        serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
 
-        lv.setAdapter(new ArrayAdapter<>(this,
-                android.R.layout.simple_list_item_1, contacts));
+        fillDatabaseWithDemoData();
+
+        try {
+            lv.setAdapter(new ContactListAdapter(this, serviceLocator.getDatabase().getAllContacts()));
+        } catch (final IOException e) {
+            e.printStackTrace();
+        }
 
     }
 
     /**
-     * Temporary method for filling the listview with some friends.
+     * Temporarily fill the database with demo data for development.
      */
-    private void getContacts() {
-        contacts.add("Eric");
-        contacts.add("Kilian");
-        contacts.add("Joost");
-        contacts.add("Stas");
+    private void fillDatabaseWithDemoData() {
+        final IDatabase database = serviceLocator.getDatabase();
+        try {
+
+            final Contact a = new Contact("Eric", new SimpleKey("jdfs09jdfs09jfs0djfds9jfsd0"));
+            final Contact b = new Contact("Stas", new SimpleKey("4ji395j495i34j5934ij534i"));
+            final Contact c = new Contact("Joost", new SimpleKey("dnfh4nl4jknlkjnr4j34klnk3j4nl"));
+            final Contact d = new Contact("Kilian", new SimpleKey("sdjnefiniwfnfejewjnwnkenfk32"));
+            final Contact e = new Contact("Cornel", new SimpleKey("nr23uinr3uin2o3uin23oi4un234ijn"));
+            if(!database.getAllContacts().isEmpty()) {
+                database.deleteContact(a);
+                database.deleteContact(b);
+                database.deleteContact(c);
+                database.deleteContact(d);
+                database.deleteContact(e);
+            }
+            database.addContact(a);
+            database.addContact(b);
+            database.addContact(c);
+            database.addContact(d);
+            database.addContact(e);
+        } catch (final IOException e) {
+            e.printStackTrace();
+        }
     }
 }
+
+/**
+ * An Adapter which converts a list with contacts into List entries.
+ */
+class ContactListAdapter extends ArrayAdapter<Contact> {
+
+    private final static int MAX_SNIPPET_SIZE = 30;
+    /**
+     * Create and initialize a ContactListAdapter.
+     *
+     * @param context the Context where the ListView is created
+     * @param contacts the list with contacts
+     */
+    ContactListAdapter(final Context context, final List<Contact> contacts) {
```

Comment:
Can you make this class final, assuming you don't want to extend this class later on?